### DISCUSSION TITLE
Not understanding `open`

### DISCUSSION THREAD

**Discussion by InusualZ** [nushell/nushell#4132] [Q&A] [answered]
Thanks to Handmade Seattle, I came across this shell. I think it's a wonderful thing.

So, I started to hack a few things together to get myself familiarized with shell. But, I came across this error which I don't know what I'm doing wrong.

I have a few `.txt` files which basically are `csv` files but with the separator being `|`. I got a strange error

![image](https://user-images.githubusercontent.com/2200611/142666220-6e04b668-fe8e-4cbb-b23a-9dd134f53f42.png)

I even tried to read the file as `--raw`

![image](https://user-images.githubusercontent.com/2200611/142667511-1b60c0bc-659f-4284-9d37-4d577804d6ab.png)

I also tried without the `from csv` and doing it manually, For some reason that I don't understand this works but only for the first file in the `ls`
![image](https://user-images.githubusercontent.com/2200611/142667755-9b577a11-af7a-4e32-adc7-07418c37968c.png)

![image](https://user-images.githubusercontent.com/2200611/142668330-34c8e8d9-916d-4447-a5d3-04b761e28be4.png)

Does anyone have any insight of why is this happening?

**Comment by sophiajt** [comment, marked as answer]
@InusualZ - thanks!

My hunch is that the file encoding of the text files is possibly throwing off the importer. There's some file encoding support in `open` but you have to know ahead of time what format you want to import from.

If you do just the `open $it.name` step and not the others, do you get string output or binary output? That might be one way to check if it's an encoding issue (or open it up in a text editor that supports telling you what the file encoding is)

**Reply by InusualZ** [reply]
Yep, it read it as binary
![image](https://user-images.githubusercontent.com/2200611/142670183-09d91aa6-9c28-4d3f-a1bc-fb5e5d8385d2.png)

By specifying the encoding did trick
![image](https://user-images.githubusercontent.com/2200611/142671109-9c5bcdce-369b-486e-a4c9-d918804f47a0.png)

Weirdly enough the `from csv` did not work, But in this case I can make it work with `lines` because I know that the csv file doesn't escape `newlines` characters.

![image](https://user-images.githubusercontent.com/2200611/142671382-04e457f1-5795-46fb-a1cf-6b5db44b9e57.png)

Thanks for the fast response :)

**Comment by sophiajt** [comment]
@InusualZ - yeah, the `from csv` gets called automatically by `open` based on the file extension. If you want just the raw text for doing things like `lines`, you'll need the `--raw` flag on `open`.
